v2/provisioned-resource_test.go

```diff
@@ -0,0 +1,49 @@
+package batcher_test
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	gobatcher "github.com/plasne/go-batcher/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetCapacity(t *testing.T) {
```

Comment:
This test feels very awkward.

The "events" var that gets loaded in the listener and then asserted outside of the listener isn't clear what or why it's doing that. The use of a WaitGroup to artificially force the code under test into submission also feels wrong.

Might I propose instead:
1. Instance the component under test with a given capacity.
2. Wire in a mock listener. Set an expectation on the mock listener that the capacity event would be fired exactly once.
3. Start it.
4. Spin until it is running, and then spin until capacity reported is what we'd expect. Set a relatively short timeout for these, which would represent test failures if the timeout is reached.
5. Set a new capacity.
6. Spin again until capacity reported is correct. Use a short timeout again to trigger test failure.
7. Assert the expectations on the mock listener.

This pattern would read much more cleanly, and would more closely resemble how this component would actually be expected to be used.

Reply:
I am not really sure how that is different. If the issue is just the events var, then I can take that out and instead set an expected value that the listener asserts against. However, "spin until running" or "spin until capacity is reported" I would still implement with WaitGroups.

Reply:
Would you consume this API using a WaitGroup?

If so, then maybe it shouldn't be async like this?

If not, then my feedback is that the test (which is a consumer) should be required to do so.

Reply:
When I consume the API, I wouldn't verify that SetCapacity at all. I would assume that if I said the capacity was x, that the Batcher did what it was supposed to do. In fact, with ProvisionedResource there is really no need to verify anything as a consumer because you always have max capacity. However, that doesn't mean the unit test shouldn't verify it.

Reply:
The test has been rewritten to drop all expectation of async.